core/string/ustring.cpp

```diff
@@ -54,34 +54,38 @@
 
 static const int MAX_DECIMALS = 32;
 
-static _FORCE_INLINE_ bool is_digit(char32_t c) {
+const char CharString::_null = 0;
+const char16_t Char16String::_null = 0;
+const char32_t String::_null = 0;
+
+bool is_digit(char32_t c) {
 	return (c >= '0' && c <= '9');
 }
 
-static _FORCE_INLINE_ bool is_hex_digit(char32_t c) {
+bool is_hex_digit(char32_t c) {
 	return (is_digit(c) || (c >= 'a' && c <= 'f') || (c >= 'A' && c <= 'F'));
 }
 
-static _FORCE_INLINE_ bool is_upper_case(char32_t c) {
+bool is_upper_case(char32_t c) {
 	return (c >= 'A' && c <= 'Z');
 }
 
-static _FORCE_INLINE_ bool is_lower_case(char32_t c) {
+bool is_lower_case(char32_t c) {
 	return (c >= 'a' && c <= 'z');
 }
 
-static _FORCE_INLINE_ char32_t lower_case(char32_t c) {
-	return (is_upper_case(c) ? (c + ('a' - 'A')) : c);
+bool is_alphabetic(char32_t c) {
+	return is_lower_case(c) || is_upper_case(c);
 }
 
-const char CharString::_null = 0;
-const char16_t Char16String::_null = 0;
-const char32_t String::_null = 0;
-
 bool is_symbol(char32_t c) {
 	return c != '_' && ((c >= '!' && c <= '/') || (c >= ':' && c <= '@') || (c >= '[' && c <= '`') || (c >= '{' && c <= '~') || c == '\t' || c == ' ');
 }
 
+char32_t lower_case(char32_t c) {
+	return (is_upper_case(c) ? (c + ('a' - 'A')) : c);
+}
```

Comment:
That only works for ASCII as far as I know, so it's a bit weak to have as a public API.

Reply:
This is for ASCII only, for the Node creation dialog it's probably sufficient.

For general purpose usage, we should add a simplified version of the `TextServer::shaped_text_update_breaks`, but without shaping/BiDi.

Reply:
Then maybe it should be moved to the CreateDialog to [keep solutions local](https://docs.godotengine.org/en/latest/community/contributing/best_practices_for_engine_contributors.html#solutions-must-be-local)?

Reply:
And if it's necessary to have Unicode aware `to_upper`, `to_lower`, `to_title`, ICU has these as well.

Reply:
> Then maybe it should be moved to the CreateDialog

`lower_case` is not used anywhere and too simple to be a separate function, probably should be removed completely.

Reply:
Also, I do not see any reason for removing `_FORCE_INLINE_`, most compilers probably do not care about this attribute anyway, but it won't hurt to keep it.

Reply:
> `lower_case` is not used anywhere and too simple to be a separate function, probably should be removed completely.

Did not notice it's already was there.

Probably better to just replace `if (is_upper_case(c) || is_digit(c)) {` in the `CreateDialogCandidate` with inlined check and do not change `ustring` at all, we already have duplicates of this ASCII only functions in some other places.

Reply:
> we already have duplicates of this ASCII only functions in some other places.

I saw that too and thought exposing these functions would be useful in those places as well. 
![image](https://user-images.githubusercontent.com/25907608/147143587-8e4eeef2-0cab-4bd6-a3a2-f30507497c96.png)

ustring seems like a collection of string classes and functions, and these few ascii functions seem like they fit. But perhaps it would be better to put these functions in a separate file like `ascii.h`?

I'll add the inline keyword back in, probably got removed accidentally.